native_client/deepspeech.cc

```diff
@@ -429,7 +478,37 @@ ModelState::decode(vector<float>& logits)
     inputs.data(), n_frames, num_classes, *alphabet, beam_width,
     cutoff_prob, cutoff_top_n, scorer);
 
-  return strdup(alphabet->LabelsToString(out[0].tokens).c_str());
+  return out;
+}
+
+Metadata* ModelState::decode_metadata(vector<float>& logits) 
+{
+  vector<Output> out = decode_raw(logits);
+
+  Metadata* metadata = (Metadata*)malloc(sizeof (Metadata));
+  metadata->num_items = out[0].tokens.size();
+  metadata->items = (MetadataItem*)malloc(sizeof(MetadataItem) * metadata->num_items);
+
+  // Loop through each character
+  for (int i = 0; i < out[0].tokens.size(); ++i) {
+    char* character = (char*)alphabet->StringFromLabel(out[0].tokens[i]).c_str();
+
+    // Note: 1 timestep = 20ms
+    float start_time = static_cast<float>(out[0].timesteps[i] * AUDIO_WIN_STEP);
+
+    MetadataItem item;
```

Comment:
what's the lifetime of this `item` ? I'm unsure about how it is allocated right now

Reply:
It should stick around until you free the items array on the Metadata parent.

Reply:
But it's allocated on the stack, while `Metadata` is on the heap

Reply:
@lissyx Good catch!